### PR TITLE
Inverted index api

### PR DESCRIPTION
**Pull request by tomipaul**
InvertedIndex class, unit tests, api endpoints and integration tests

### CODE REVIEW

File: tests/inverted-index-test.js [hunk 1]
```diff
@@ -121,38 +124,119 @@ describe('Inverted index class creates index', () => {
...
+      expect(searchResult).toEqual('Invalid index Object');
+    });
+
+    it('returns a message if index is an empty object', () => {
+      const searchResult = invertedIndex.searchIndex({},
+      'anotherValidFile.json');
+      expect(searchResult).toEqual('Invalid index Object');
     });
   });
 });
```
Comment (rkterungwa16) [inline tests/inverted-index-test.js:277]: Good job Impressive test coverage.  But can't seem to find search for words having other characters attached to them. Example: 'what#$@'

Comment (tomipaul) [reply tests/inverted-index-test.js:277]: normalize handles that. normalize is a static method of the InvertedIndex class. It is called within the searchIndex method to normalize the search terms: 
-It converts non-word characters to spaces, 'what#$@' becomes 'what   '
-Then it trims boundary spaces, 'what   ' becomes 'what'
-And finally, it converts the result to lowercase. 'what' is already lowercase.
I will include the test to show its functionality in the searchIndex method

File: dist/inverted-index.js [hunk 2]
```diff
@@ -1,230 +0,0 @@
...
-        }
-      });
-      return bookContainsAll;
-    }
-  }]);
-
-  return InvertedIndex;
-}();
-
-exports.default = InvertedIndex;
```
Comment (EmaEvidence) [inline dist/inverted-index.js:230]: I think you are not to upload your 'dist' folder.

**Review by EmaEvidence — COMMENTED** [review]
Good Job, your code is readable and you adhered to the style guide.

File: dist/inverted-index.js (hunk 2 above)
Comment (tomipaul) [reply dist/inverted-index.js:230]: Yes, I have added it to .gitignore already but it's already being tracked. I will make neccesary corrections.

**Pull request merged**